docs/user_guide/config.rst

```diff
@@ -49,19 +49,55 @@ multiple external repositories can be used as the example below illustrates for
       url: https://github.com/IAMconsortium/common-definitions.git/
   definitions:
     region:
-      repository: common-definitions
+      repository:
+        name: common-definitions
     variable:
       repositories:
-        - common-definitions
-        - legacy-definitions
+        - name: common-definitions
+        - name: legacy-definitions
   mappings:
-    repository: common-definitions
+    repository:
+      name: common-definitions
 
-The value in *definitions.region.repository* needs to reference the repository in the
-*repositories* section.
+The value in *definitions.region.repository* can be a list or a single value, needs to
+contain the ``name`` keyword and reference the repository in the *repositories* section.
 
 For model mappings the process is analogous using *mappings.repository*.
 
+Filter code lists imported from external repositories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since importing the entirety of, for example, common-definitions is too much for most
+projects, the list can be filtered using ``include`` and ``exclude`` keywords. Under
+these keywords, lists of filters can be given that will be applied to the code list from
+the given repository.
+
+The filtering can be done by any attribute:
+
+.. code:: yaml
+
+  repositories:
+    common-definitions:
+      url: https://github.com/IAMconsortium/common-definitions.git/
+  definitions:
+    variable:
+      repository:
+        name: common-definitions
+        include:
+          - name: [Primary Energy*, Final Energy*]
+          - name: "Population*"
+            tier: 1
+        exclude:
+          - name: "Final Energy|*|*"
```

Comment:
This is a misleading example - it seems to show only level-2 exclusion but it excludes all variables at level 2 or below. Also, pyam excludes the filter-String from the level-calculation. Better to use **pyam.utils.pattern_match** for consistency with pyam-filtering and support the level-argument explicitly.
```suggestion
          - name: "Final Energy|*"
             level: "1+"
```